wpilibc/src/main/native/cpp/LEDPattern.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "frc/LEDPattern.h"
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <numbers>
+
+#include <wpi/MathExtras.h>
+#include <wpi/timestamp.h>
+
+using namespace frc;
+
+LEDPattern::LEDPattern(const LEDPatternFn& impl) : m_impl(std::move(impl)) {}
```

Comment:
Could someone more versed with C++ confirm whether `std::move`ing a const lvalue reference makes sense? Right now I only see constructing an `LEDPattern` from a lambda, so `LEDPattern(LEDPatternFn&& impl)` would probably also work.

Reply:
No, this isn't correct to do.  It is if impl is passed by value, but not by const ref.